### PR TITLE
Sample of using build info to generate keystrokes

### PR DESCRIPTION
**Pull request by cbbrowne**
FYI, I split off the date into QMK_BUILDDATE so it is separately accessible

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -114,7 +114,7 @@ include $(TMK_PATH)/protocol/lufa.mk
 include $(TMK_PATH)/common.mk
 include $(TMK_PATH)/rules.mk
 
-GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags 2>/dev/null || date +"%Y-%m-%d")
-
+GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags 2>/dev/null)
+BUILD_DATE := $(shell date +"%Y-%m-%d-%H:%M:%S")
```
Comment (algernon) [inline Makefile]: There's one issue I see with this: if the firmware is not built from a git checkout, `QMK_VERSION` will be empty. I'd suggest keeping the new `BUILD_DATE`, but preserve the former behaviour of `QMK_VERSION` too, by having an `|| echo ${BUILD_DATE}` at the end of `GIT_VERSION`.

This way we get the best of both.

Comment (jackhumbert) [inline Makefile]: I agree!

**Comment by jackhumbert** [comment]
If you wanna make the change that @algernon suggested, I'll get this merged!

**Comment by jackhumbert** [comment]
Re the compiled.hex files, we're including those so that new users have a reference if they are unable to build, or just want to try out a keymap, they can without compiling. This is something the Ergodox EZ did that we recently made part of the whole repo.

**Comment by cbbrowne** [comment]
These all seem like reasonable changes, but **please** review the intent...

- I have removed many of the .hex files from the repo as they are properly generated code that I imagine shouldn't be included.  Please validate that this is the right idea.

- Similarly, added .gitignore rules for the .build directories, and for *many* (not yet all) of the .hex files

If this is the right idea, then there probably should be a .gitignore rule up at a fairly high level indicating that *.hex files should be left out.

**Comment by jackhumbert** [comment]
```
/*.hex
/keyboard/**/*.hex
/keyboard/**/keymaps/**/*.hex
!/keyboard/**/keymaps/**/compiled.hex
```

Are the rules I have to leave out the unnecessary .hex files - compiled.hex should still be included. I'm not sure how that `planck_experimental.hex` made its way through.

**Comment by cbbrowne** [comment]
There, that should be better; I reverted the compiled.hex files

**Comment by jackhumbert** [comment]
Awesome - thanks!

**Pull request merged**